CRM/Event/Cart/Form/Checkout/Payment.php

```diff
@@ -774,17 +785,20 @@ protected function apply_discount($discountCode, &$price_set_amount, &$cost) {
           $active2 = TRUE;
         }
       }
-      if ($discountValue['is_active'] == TRUE && ($discountValue['count_max'] == 0 || ($discountValue['count_max'] > $discountValue['count_use'])) && $active1 == TRUE && $active2 == TRUE) {
+      if ($discountValue['is_active'] == TRUE && ($discountValue['count_max'] == 0 || ($discountValue['count_max'] > $discountValue['count_use'])) && $active1 == TRUE && $active2 == TRUE && $event_match == TRUE) {
         foreach ($price_set_amount as $key => $price) {
           if (array_search($price['price_field_value_id'], $discounted_priceset_ids) != NULL) {
             $discounted = _cividiscount_calc_discount($price['line_total'], $price['label'], $discountValue, $autoDiscount, "USD");
             $price_set_amount[$key]['line_total'] = $discounted[0];
             $cost += $discounted[0];
             $price_set_amount[$key]['label'] = $discounted[1];
+	  } else {
+            $cost += $price['line_total'];
           }
         }
+        $stat = TRUE;
       }
     }
+    return $stat;
   }
-
```

Comment:
I think this will be another jenkins style issue - it requires a blank like at the end of the class